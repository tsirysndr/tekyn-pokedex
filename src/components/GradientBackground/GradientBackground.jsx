import React from "react";
import { View } from "react-native";
import { LinearGradient } from "expo-linear-gradient";
import styled from "@emotion/native";
import PropTypes from "prop-types";

const Container = styled(LinearGradient)`
  flex: 1;
  align-items: center;
  justify-content: center;
`;

export default function GradientBackground({ children }) {
  return <Container colors={["#F4A261", "#141A2B"]}>{children}</Container>;
}

GradientBackground.defaultProps = {
  children: null,
};

GradientBackground.propTypes = {
  children: PropTypes.node,
};
