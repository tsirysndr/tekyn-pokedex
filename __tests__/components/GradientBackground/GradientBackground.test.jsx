import GradientBackground from "@Components/GradientBackground";
import { render } from "@testing-library/react-native";

test("renders the GradientBackground", () => {
  const tree = render(<GradientBackground />);
  expect(tree).toMatchSnapshot();
});
