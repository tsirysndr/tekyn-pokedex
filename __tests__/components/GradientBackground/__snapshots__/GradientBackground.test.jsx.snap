// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the GradientBackground 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4294222433,
        4279507499,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;
